### PR TITLE
Add functionality: Merges BitOps Defaults folder with targetted Helm Chart

### LINKED ISSUE #9: Copy Defaults - undefined env vars ERROR and NC
https://github.com/bitovi/bitops/blob/master/scripts/helm/copy-defaults.sh

ERROR and NC seem to be undefined.

Need to confirm they're undefined or, if not, comment where the values are coming from.

### PR DESCRIPTION
**Pull request by PhillypHenning**
fixes https://github.com/bitops-plugins/helm/issues/9

### CODE REVIEW

**Review by LeoDiazL — COMMENTED** [review]
Everything looks good to me.

File: scripts/copy-defaults.sh [hunk 1]
```diff
@@ -8,66 +8,40 @@ HELM_CHART="$1"
...
-if [ "1" == "1" ]; then
-# if [ "$(shyaml get-value copy_defaults.crds < "$HELM_BITOPS_CONFIG")" == 'True' ]; then
-    echo "COPY_DEFAULT_CRDS set"
-    if [ -d "$DEFAULT_HELM_CHART_DIRECTORY/crds" ]; then
-        echo "default crds/ exist"
-        cp -rf "$DEFAULT_HELM_CHART_DIRECTORY/crds/." "$HELM_CHART_DIRECTORY/crds/"
-    else
-        printf "${ERROR} crds/ does not exist...${NC}"
-    fi
+printf "COPY_DEFAULT_CRDS set..."
```
Comment (arm4b) [inline scripts/copy-defaults.sh]: I noticed printf doesn't have line return `\n`, meaning next print or echo will print on the same line. Is it by intent?

File: scripts/copy-defaults.sh [hunk 2]
```diff
@@ -8,66 +8,40 @@ HELM_CHART="$1"
...
+    cp -rf "$DEFAULT_HELM_CHART_DIRECTORY/crds/." "$HELM_CHART_DIRECTORY/crds/"
+    ls $HELM_CHART_DIRECTORY/crds/
 else
     echo "COPY_DEFAULT_CRDS not set"
 fi
 
-# Copy default Charts.
-
-if [ "1" == "1" ]; then
-# if [ "$(shyaml get-value copy_defaults.charts < "$HELM_BITOPS_CONFIG")" == 'True' ]; then
```
Comment (arm4b) [inline scripts/copy-defaults.sh:31]: Looking at this, I guess 1 == 1 was just a placeholder for the future non-implemented yet logic with shyaml reading the values from the yaml file.

I'm good for us removing it, if there was no clear `TODO: Comment` explaining the future idea.

File: deploy.sh [hunk 3]
```diff
@@ -23,9 +23,13 @@ echo "calling helm/deploy ..."
 # if subdirectory is not provided, iterate subdirectories
 if [ -z "$ENVIRONMENT_HELM_SUBDIRECTORY" ]; then
   echo "ENVIRONMENT_HELM_SUBDIRECTORY not provided, iterate all helm charts in $ENV_DIR/helm"
+
   for helm_chart_dir in $HELM_ROOT_OPERATIONS/*/; do
     helm_chart_dir=${helm_chart_dir%*/}     # remove the trailing "/"
     helm_chart_dir=${helm_chart_dir##*/}    # get everything after the final "/"
+    if [ "$helm_chart_dir" == "$BITOPS_DEFAULT_ROOT_DIR" ] || [ "$helm_chart_dir" == "$DEFAULT_ROOT_DIR" ]; then
+      continue
```
Comment (arm4b) [inline deploy.sh:33]: I understand the PR here is to fix the $ERROR and $NC cases.

Any context about skipping the loop iteration in this case? as it changes some logic.

**Review by arm4b — COMMENTED** [review]
Left a couple of questions and notes so I can understand the PR better.

> Copy defaults #10

Can we have a more detailed title for this PR?
Because it's something that we rely on when creating a [Release Changelog](https://github.com/bitovi/bitops/releases), it's worth spending a bit more time to make a PR titles better.

File: deploy.sh (hunk 3 above)
Comment (arm4b) [reply deploy.sh:33]: I see now it was discussed in [Slack](https://bitovi.slack.com/archives/C0129U3GL93/p1667580285383069?thread_ts=1667576988.091839&cid=C0129U3GL93).

@PhillypHenning please include a quick explanation as a code comment, - that'll make it clear for everyone.
Two questions on the same piece of code is definitely an indicator to document more details in the code.

**Review by arm4b — APPROVED** [review]
Thanks!

**Pull request merged**